frontend/main.js

```diff
@@ -49,6 +49,16 @@ console.info('CONTRACTS_VERSION:', process.env.CONTRACTS_VERSION)
 console.info('LIGHTWEIGHT_CLIENT:', process.env.LIGHTWEIGHT_CLIENT)
 console.info('NODE_ENV:', process.env.NODE_ENV)
 
+if (process.env.CI) {
+  const originalFetch = self.fetch
+  self.fetch = (...args) => {
+    return originalFetch.apply(self, args).catch(e => {
+      console.error('FETCH FAILED', args, new Error().stack, e)
+      throw e
+    })
```

Comment:
For both of these I don't think you need to do `new Error().stack`, you can just use `e.stack`.

Reply:
No, they are distinct values. `e.stack` will not have information about the caller, which is needed to figure out the offending location. If `e.stack` were enough, we wouldn't even need this code, since the exception will appear in the console.